Adds the non zero prefixed MRN database constraint

fredkingham opened the pull request:
Adds a database constraint to make it so that it is impossible to change an MRN so that it has a zero prefix.

Tested with csv imports and bulk updates.

Commit:
Adds the non zero prefixed MRN database constraint

Adds a database constraint to make it so that it is impossible to change
an MRN so that it has a zero prefix.

davidmiller commented:
Does the constraint work if we deploy it to a database that hasn't yet been cleansed of 0prefixes? 
(If not, does this need to point at a different branch... )

davidmiller commented on elcid/models.py:
```diff
@@ -75,6 +75,12 @@ def age(self):
         verbose_name_plural = "Demographics"
+        constraints = [
+            models.CheckConstraint(
+                check = ~models.Q(hospital_number__startswith=0),
```
I think I landed at either a different spelling or a comment explaining how _bitwise_ not works here...

Commit:
Adds a comment to elcid models

Negation of a Q expression is done with a tilda ('~') this looks wrong
at first glance so a comment has added with a link to the django
documentation to explain the syntax.
